02_lecture/homework/task2.ts

```diff
@@ -0,0 +1,15 @@
+function arrayMutateRemove<Type>(
+  arr: Array<Type>,
+  func: Function
+): Array<Type> {
+  const removedElements: Array<Type> = [];
+  for (let i = 0; i < arr.length; i++) {
+    const isRemoved = func.call(this, arr[i]);
+    if (isRemoved) {
+      removedElements.push(arr[i]);
+      arr.splice(i, 1);
+      i = i - 1;
+    }
+  }
+  return removedElements;
```

Comment:
I'd avoid additional variables and native "for" loops. It will simplify code base
```suggestion
    array.reduce(
    (acc, currentElement, index) =>
      predicate(currentElement, index)
        ? [...acc, array.splice(index, 1)[0]]
        : acc,
    [] as TArrayElement[],
  );
```

Reply:
done